src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1472,7 +1472,7 @@ public static IEnumerable<int> Range(int start, int count)
 
         private static IEnumerable<int> RangeIterator(int start, int count)
         {
-            for (int i = 0; i < count; i++) yield return start + i;
+            for (int end = start + count; start != end; start++) yield return start;
```

Comment:
A negative count getting to here would be a bug in either version, as the calling method is specified as throwing in that case.